internal/exec/stages/files/files.go

```diff
@@ -164,6 +152,11 @@ func (tmp dirEntry) create(l *log.Logger, u util.Util) error {
 			return err
 		}
 
+		uid, gid, err := u.ResolveNodeUidAndGid(d.Node, 0, 0)
```

Comment:
mention this refactor in the commit message or split into one commit for strictly fixing for the pointer types and one for the refactor